test/compilable/ccompile.d

```diff
@@ -0,0 +1,36 @@
+
+/* REQUIRED_ARGS: -O
+ */
+
+// Adapted from DMC++ test file test3/ccompile.c
+
+
+    struct HDS        {
+      char state;
+      uint done;
+      uint retry;
+      uint[15] tests;
+   }
+
+void funchds(char *p_adults)
+{
+   int cupx, chemx;
+   HDS *p_cup;
+
+   for (cupx = 1, p_cup = null; cupx <=  48 ; cupx ++, p_cup ++)
+   {
+         for (chemx = 0; chemx <  15 ; chemx++)
+         {
+            if (p_cup.done) {
```

Comment:
Can't the optimizer make this function a single `ud2` instruction because of this null pointer access? Maybe better make `p_cup` a function argument?

Reply:
This test crashes the compiler when it is compiled. It was never intended to be runnable code, it was the minimized test case for a crashing bug decades ago.